devops/site.yml

```diff
@@ -0,0 +1,79 @@
+- name: Launch the ec2 instance
+  hosts: localhost
+  connection: local
+  gather_facts: false
+  become: false
+  vars:
+    aws_access_key: "{{ lookup('env', 'AWS_ACCESS_KEY_ID') }}"
+    aws_secret_key: "{{ lookup('env', 'AWS_SECRET_ACCESS_KEY') }}"
+
+  roles:
+    - role: ec2
+
+  post_tasks:
+    - name: Register the instance to dynamic inventory
+      ec2_remote_facts:
+        filters:
+          instance-state-name: 'running'
+          "tag:Name": "{{ ec2_instance_name }}"
+        region: "{{ ec2_region }}"
+      register: ec2
+
+    - name: Add new instance to jupyterhub & geoserver groups
+      add_host:
+        hostname: "{{ item.public_dns_name }}"
+        groups: "jupyter,geoserver"
+        ansible_ssh_private_key_file: "{{ ec2_keypair_path }}"
+        ansible_ssh_user: "{{ ansible_default_user }}"
+      with_items: '{{ec2.instances}}'
+
+
+# Handle installing python 2.7.11 on 16.04
+- name: Hack to resolve 16.04 python2/3 issue
+  hosts: geoserver
+  gather_facts: false
+  tasks:
+    - name: Install python2
+      raw: >-
+        sudo apt-get update && sudo apt-get install -y python
+
+
+# Deploy geoserver
+- include: geoserver/deploy.yml
+
+
+- name: Provisiong Jupyter Hub, install geonotebook
+  hosts: jupyter
+  become: yes
+  become_user: root
+  roles:
+    - role: jupyterhub
+      jupyterpw: "{{ jupyter_user_password }}"
+    - role: geonotebook
+      geonotebook_version: ansible-ec2-deploy
+
+  post_tasks:
+    - name: Create example directory
+      file:
+        path: "/home/{{ item.username }}/examples"
+        state: directory
+        owner: "{{ item.username }}"
+        group: "{{ item.username }}"
+      with_items: "{{ notebook_users }}"
+      when: notebook_users is defined
+
+    - name: Copy example notebooks
+      shell: >-
+        cp /opt/geonotebook/notebooks/*.ipynb /home/{{ item.username }}/examples/
```

Comment:
Using fileglob or copy with ```remote_src: true``` ends up being more confusing than helpful,  we're already looping over users so we can't also loop over fileglobs without deglobbing them in a different task.